fractional compositions in AtomicOrbitals

dyllamt opened the pull request:
## Summary

- add "sanitation" step to `AtomicOrbitals` composition featurizer
- fractional compositions (Na0.5Cl0.5) are now supported
- added test to ensure equivalent compositions (NaCl v.s. Na0.5Cl0.5) return the same features

## TODO
- possibly update pymatgen.core.MolecularOrbitals to require integer subscripts

dyllamt commented:
## Update
pymatgen's `MolecularOrbitals` were updated to raise a `ValueError` if non-integer subscripts are passed (`AtomicOrbitals` calls this pymatgen class to generate features). The sanitation step of this featurizer ensures that only integer value subscripts are passed to pymatgen.

WardLT commented:
Thanks @dyllamt !

This looks good to me. Anyone want to review before I merge? I have a user who wants to use this feature.

PS. Adding that this Fixes #301

computron commented:
The default max_denominator for ``get_integer_formula_and_factor`` is 10,000. @dyllamt does that sounds reasonable? Or do you want to make it something like 1000. I don't know how well the code performs with for example a few thousand atoms in the Composition.

dyllamt commented:
The performance is not much of an issue (~7ms per featurize call with 10,000 atoms), but the discussion does raise an issue that using `get_integer_formula_and_factor` will truncate a composition when there is a **very** dilute element (e.g. FeC0.00001 --> Fe).

I think this is ok behavior from a scientific standpoint, since if the element is that dilute, it is not contributing significant orbital character. But it seems like the user should still be aware of this behavior, so I added a warning in the docstring and a warning message in the code.

## Summary
* dilute compositions (FeC0.00001) are truncated (becomes Fe) by `get_integer_formula_and_factor`
* updated docstring to alert user to this behavior
* added warning message for when this occurs (using `warnings.warn`)
* added test for the warning message

computron commented:
Thanks!
